src/Symfony/Component/Ldap/Adapter/ExtLdap/Query.php

```diff
@@ -60,7 +60,20 @@ public function execute()
 
             $con = $this->connection->getResource();
 
-            $this->search = @ldap_search(
+            switch ($this->options['scope']) {
+                case static::SCOPE_BASE:
+                    $func = 'ldap_read';
+                    break;
+                case static::SCOPE_ONELEVEL:
+                    $func = 'ldap_list';
+                    break;
+                case static::SCOPE_SUBTREE:
+                default:
```

Comment:
The default should actually throw an exception (which should actually never be thrown, as the OptionsResolver is actually the one which sanitizes the `scope` option) instead of being a fall-through for the `subtree` scope.